soh/soh/Enhancements/mods.cpp

```diff
@@ -486,6 +488,153 @@ void RegisterBonkDamage() {
+    CHEST_BOMB_TRAP,
+    CHEST_VOID_TRAP,
+    CHEST_AMMO_TRAP,
+    CHEST_KILL_TRAP,
+    CHEST_TELEPORT_TRAP,
+    CHEST_TRAP_MAX
+} AltTrapType;
+
+const char* altTrapTypeCvars[] = {
+    "gChestTraps.Ice",
+    "gChestTraps.Burn",
+    "gChestTraps.Shock",
+    "gChestTraps.Knock",
+    "gChestTraps.Speed",
+    "gChestTraps.Bomb",
+    "gChestTraps.Void",
+    "gChestTraps.Ammo",
+    "gChestTraps.Kill",
+    "gChestTraps.Tele"
+};
+
+std::vector<AltTrapType> getEnabledChestTraps () {
+    std::vector<AltTrapType> enabledChestTraps;
+    for (int i = 0; i < CHEST_TRAP_MAX; i++) {
+        if (CVarGetInteger(altTrapTypeCvars[i], 0)) {
+            enabledChestTraps.push_back(static_cast<AltTrapType>(i));
+        }
+    }
+    return enabledChestTraps;
+};
+
+void RegisterAltChestTypes() {
+    static AltTrapType roll = CHEST_TRAP_MAX;
+    static int statusTimer = -1;
+    static int eventTimer = -1;
+
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnItemReceive>([](GetItemEntry itemEntry) {
+        if (!CVarGetInteger("gChestTraps.enabled", 0) || itemEntry.itemId != RG_ICE_TRAP) {
+            return;
+        }
+        roll = RandomElement(getEnabledChestTraps());
+        switch (roll) {
+            case CHEST_ICE_TRAP:
+                GameInteractor::RawAction::FreezePlayer();
+                break;
+            case CHEST_BURN_TRAP:
+                GameInteractor::RawAction::BurnPlayer();
+                break;
+            case CHEST_SHOCK_TRAP:
+                GameInteractor::RawAction::ElectrocutePlayer();
+                break;
+            case CHEST_KNOCK_TRAP:
+                eventTimer = 3;
+                break;
+            case CHEST_SPEED_TRAP:
+                Audio_PlaySoundGeneral(NA_SE_VO_KZ_MOVE, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                GameInteractor::State::RunSpeedModifier = -2;
+                statusTimer = 200;
+                Overlay_DisplayText(10, "Speed Decreased!");
+                break;
+            case CHEST_BOMB_TRAP:
+                eventTimer = 3;
+                break;
+            case CHEST_VOID_TRAP:
+                Audio_PlaySoundGeneral(NA_SE_EN_GANON_LAUGH, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                eventTimer = 3;                    
+                break;
+            case CHEST_AMMO_TRAP:
+                eventTimer = 3;
+                Overlay_DisplayText(5, "Ammo Halved!");
+                break;
+            case CHEST_KILL_TRAP:
+                GameInteractor::RawAction::SetPlayerHealth(0);
+                break;
+            case CHEST_TELEPORT_TRAP:
+                eventTimer = 3;
+                break;
+        }
+    });
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnPlayerUpdate>([]() {
+        Player* player = GET_PLAYER(gPlayState);
+        if (statusTimer == 0) {
+            GameInteractor::State::RunSpeedModifier = 0;
+            statusTimer--;
```

Comment:
It's a small thing, but since statusTimer and eventTimer are decremented in both the if and the else cases, you don't need to duplicate the decrement calls inside the if statements.  You can just put a single decrement each after the ifs at the very end of this function.

Reply:
Small but optimized in comparison, made the changes to this in the new commit